### PR TITLE
Add support for tasks and ES Modules

### PR DESCRIPTION
**Pull request by abendigo**
### 🤔 What's changed?

The project I am using screenplay.js in has "type": "module". Because of that, tasks would not load

### ⚡️ What's your motivation? 

I like the promise that screenplay.js makes, sub-second tdd. But, as a gluton for punishment, I also like using ES modules.

### 🏷️ What kind of change is this?

- :zap: New feature (non-breaking change which adds new behaviour)

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [ ] I've changed the behaviour of the code
  - [ ] I have added/updated tests to cover my changes.
- [X] My change requires a change to the documentation.
  - [X] I have updated the documentation accordingly.
- [X] Users should know about my change
  - [x] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,12 +5,12 @@
 Cucumber Screenplay is a small library for [Cucumber.js](https://github.com/cucumber/cucumber-js/) that enables better
```
Comment (abendigo) [inline README.md:5]: I am not sure why VSCode mangled this file so badly!

Comment (aslakhellesoy) [reply README.md:5]: Thanks for this @abendigo. Could you please undo all of the accidental edits here so your PR only contain relevant changes?

Comment (abendigo) [reply README.md]: Updated. Thanks @aslakhellesoy for taking a look! Turns out, I have "Format on Save" turned on at the user level.

Comment (abendigo) [reply README.md]: @aslakhellesoy Hey, any update on this?

File: README.md [hunk 2]
```diff
@@ -239,6 +239,31 @@ export default class World extends ActorWorld {
 setWorldConstructor(World)
 ```
 
+If you are using ES Modules (by setting `"type": "module"` in `package.json`), you need to
+change ypur `World` constructor, and add a `Begin` step:
```
Comment (mattwynne) [inline README.md:243]: What's a `Begin` step? Do you mean a `Before` hook?

File: README.md [hunk 3]
```diff
@@ -239,6 +239,31 @@ export default class World extends ActorWorld {
...
+  public startSession: StartSession
+  public shout: Shout
+  public inboxMessages: InboxMessages
+  constructor(props: IWorldOptions) {
+    super({ ...props, packageType: 'module' })
+  }
+}
+setWorldConstructor(World)
+
+Begin(() => {
```
Comment (mattwynne) [inline README.md:260]: If you're going to `await` then this function would need to be `async` right?

File: src/ActorWorld.ts [hunk 4]
```diff
@@ -2,15 +2,29 @@ import { World, IWorldOptions } from '@cucumber/cucumber'
...
+export interface IActorWorldOptions extends IWorldOptions {
+  packageType: 'commonjs' | 'module'
+}
 
 export default class ActorWorld extends World {
   public readonly actorLookup = new ActorLookup()
+  public readonly promise
+
+  constructor(props: IActorWorldOptions) {
+    console.log('>>>> ActorWorld', props)
```
Comment (mattwynne) [inline src/ActorWorld.ts:16]: Is this debug logging? Could it be removed?

File: README.md (hunk 3 above)
Comment (mattwynne) [inline README.md:260]: Would it be possible to write this hook from within the `ActorWorld` to save the user having to have this extra configuration step?

**Review by mattwynne — CHANGES_REQUESTED** [review]
I'd like to see some tests for this, particularly the `assignAsyncTasks` function.

I'm afraid I don't know the codebase well enough to advise where / how to implement those tests. Perhaps @aslakhellesoy can advise.

**Comment by mattwynne** [comment]
@abendigo thanks for submitting. If you want to chat 🍵  about my feedback, you're welcome to join us in our [community Slack](https://cucumber.io/community#slack) in the [#committers](https://cucumberbdd.slack.com/archives/C62D0FK0E) channel 😄 

I'm also live pairing ⚡ you can book a slot with me [here](https://calendly.com/mattwynne).

File: src/ActorWorld.ts (hunk 4 above)
Comment (abendigo) [reply src/ActorWorld.ts:16]: yikes! Sorry about that! Yes, it should be removed!

File: README.md (hunk 2 above)
Comment (abendigo) [reply README.md:243]: yes, I did mean Before.

File: README.md (hunk 3 above)
Comment (abendigo) [reply README.md:260]: correct! I'll update this as well.

Comment (abendigo) [reply README.md:260]: I couldn't find a way to do that, but I'd be happy to go through it in a pairing session!

File: src/ActorWorld.ts [hunk 5]
```diff
@@ -2,15 +2,29 @@ import { World, IWorldOptions } from '@cucumber/cucumber'
...
+    const { packageType = 'commonjs', ...rest } = props
 
-  constructor(props: IWorldOptions) {
-    super(props)
-    if (this.parameters.tasks) {
+    super(rest)
+
+    if (packageType == 'commonjs' && this.parameters.tasks) {
```
Comment (abendigo) [inline src/ActorWorld.ts:21]: missed the triple equals here!

**Review by aslakhellesoy — APPROVED** [review]
Looks great! Thanks for contributing this - much appreciated.

**Comment by aslakhellesoy** [comment]
Hi @abendigo,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Pull request merged**